src/message_service.py

```
@@ -0,0 +1,513 @@
+
+            # 2. Attachment Handling
+            to_download = [a for a in msg.attachments if is_supported_attachment(a)]
+            if to_download:
+                # logger.debug("[%s] Found %d attachments to download", msg_id, len(to_download))
+                # download_start = time.perf_counter()
+
+                try:
+                    # This is the most likely place for a hang
+                    downloads = await asyncio.gather(*(download_attachment(self.httpx_client, a) for a in to_download))
+                    # logger.debug("[%s] Downloads completed in %.4fs", msg_id, time.perf_counter() - download_start)
+                except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)